### PR TITLE
chore(components): support SVG title attribute for icon description

### PR DESCRIPTION
**Pull request by asudoh**
Refs #1750.

#### Changelog

**New**

- `<title>` for experimental icons (and custom icons via `renderIcon`).
- `aria-label` for experimental icons, where there are _not_ `aria-hidden`.

**NOTE**: Didn't do above for `<NumberInput>` as it's covered by: #1975
**Removed**

- `aria-label` from experimental icons, where there are `aria-hidden`.

### CODE REVIEW

File: src/components/CopyButton/CopyButton.js [hunk 1]
```diff
@@ -99,8 +99,9 @@ export default class CopyButton extends Component {
           <Copy16
             className={`${prefix}--snippet__icon`}
             aria-label={iconDescription}
-            alt={iconDescription}
-          />
+            alt={iconDescription}>
```
Comment (emyarod) [inline src/components/CopyButton/CopyButton.js]: is `alt` a valid attribute here? I think the `<title>` element is supposed to replace our invalid usage of `alt` attribute on SVGs

Comment (asudoh) [reply src/components/CopyButton/CopyButton.js]: Great catch @emyarod! - Fixed.

File: src/components/AccordionItem/AccordionItem.js [hunk 2]
```diff
@@ -138,8 +138,9 @@ export default class AccordionItem extends Component {
           {componentsX ? (
             <ChevronRight16
               className={`${prefix}--accordion__arrow`}
-              aria-label={iconDescription}
-            />
+              aria-label={iconDescription}>
```
Comment (joshblack) [inline src/components/AccordionItem/AccordionItem.js]: Do we need to double-up on `aria-label` and `title`?

Comment (asudoh) [reply src/components/AccordionItem/AccordionItem.js]: Good question @joshblack - `<title>` is for hover-over text whereas `aria-label` is for screen reader, but don't hesitate to speak up if you believe one covers another.

**Comment by emyarod** [comment]
@asudoh it looks like we there are a few conflicts that need to be resolved now

File: src/components/AccordionItem/AccordionItem.js (hunk 2 above)
Comment (joshblack) [reply src/components/AccordionItem/AccordionItem.js]: @asudoh we've gotten feedback around the tooltips themselves from `title`, might be helpful for icons to default to not having it. If someone would like a `title`, they should place it on the interactive node (like `<button>`) that it's associated with.

**Comment by joshblack** [comment]
bump @asudoh for merge conflicts, PR looks great! 👍

**Comment by asudoh** [comment]
Thanks everyone for all the review and @joshblack for pointing out the surrounding `<button>` thing! Given @vpicone worked on that area I see many of my changes here are redundant. I'm planning to sort it out next week - Which may end up with updating this PR or withdrawing this PR.

**Comment by asudoh** [comment]
Removed `<title>` in `v10` icons where there are surrounding `<button>` or `<div role="button">` with `title` attribute.

**Pull request merged**